jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStore.java

```diff
@@ -119,77 +193,68 @@ public boolean isPassivating()
         return true;
     }
 
+    public boolean isFindExpiredSession()
+    {
+        return findExpiredSession;
+    }
+
+    public void setFindExpiredSession( boolean findExpiredSession )
+    {
+        this.findExpiredSession = findExpiredSession;
+    }
+
     @Override
     public Set<String> doGetExpired( Set<String> candidates )
     {
-        if (candidates == null || candidates.isEmpty())
-        {
-            return Collections.emptySet();
-        }
-        
-        long now = System.currentTimeMillis();
-        return candidates.stream().filter( candidate -> {
+        Set<String> expired = new HashSet<>();
+
+        //check candidates that were not found to be expired, definitely they no longer exist and they should be expired
+
+        expired.addAll( candidates.stream().filter( candidate -> {
             
             if (LOG.isDebugEnabled())
+            {
                 LOG.debug( "Checking expiry for candidate {}", candidate );
-            
+            }
+            if(expired.contains( candidate ))
+            {
+                return false;
```

Comment:
This test isn't correct, it isn't checking if hazelcast considers the session expired: it could be that our node thinks it has expired, but in fact the session has been in use on some other node.